src/test/java/com/team4/testingsystem/services/TestsServiceTest.java

```diff
@@ -0,0 +1,189 @@
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.dao.EmptyResultDataAccessException;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class TestsServiceTest {
+
+    @Mock
+    Test test;
+
+    @Mock
+    Test.Builder builder;
+
+    @Mock
+    User user;
+
+    @Mock
+    UsersRepository usersRepository;
+
+    @Mock
+    TestsRepository testsRepository;
+
+    @InjectMocks
+    TestsService testsService;
+
+
+    @org.junit.jupiter.api.Test
+    void getAllSuccess() {
+        List<Test> tests = new ArrayList<>();
+        Mockito.when(testsRepository.findAll()).thenReturn(tests);
+
+        Assertions.assertEquals(tests, testsService.getAll());
+
+    }
+
+    @org.junit.jupiter.api.Test
+    void getByIdSuccess() {
+
+        Mockito.when(testsRepository.findById(1L)).thenReturn(Optional.of(test));
+
+        Assertions.assertEquals(test, testsService.getById(1L));
+
+    }
+
+    @org.junit.jupiter.api.Test
+    void getByIdFail() {
+
+        //Test doesn't exist
+        Mockito.when(testsRepository.findById(42L)).thenThrow(NoSuchElementException.class);
+
+        Assertions.assertThrows(TestNotFoundException.class, () -> testsService.getById(42L));
+
+    }
+
+    @org.junit.jupiter.api.Test
+    void createWhenAssignFail() {
+
+        //User doesn't exist
+        Mockito.when(usersRepository.findById(42L)).thenThrow(NoSuchElementException.class);
+
+        Assertions.assertThrows(UserNotFoundException.class, () -> testsService.createForUser(42L));
+    }
+
+    @org.junit.jupiter.api.Test
+    void createSuccess() {
+
+
+        Mockito.when(usersRepository.findById(1L)).thenReturn(Optional.of(user));
+
+        MockedStatic<Test> testMockedStatic = Mockito.mockStatic(Test.class);
+        testMockedStatic.when(Test::newBuilder).thenReturn(builder);
+
+        Mockito.when(builder.setUser(any())).thenReturn(builder);
+        Mockito.when(builder.setCreatedAt(any())).thenReturn(builder);
+        Mockito.when(builder.setStatus(any())).thenReturn(builder);
+        Mockito.when(builder.build()).thenReturn(test);
+
+        Mockito.when(test.getId()).thenReturn(1L);
+
+
+        Assertions.assertEquals(1L, testsService.createForUser(1L));
+
+    }
+
+    @org.junit.jupiter.api.Test
+    void startSuccess() {
+
+        Mockito.when(testsRepository.existsById(1L)).thenReturn(true);
+
+        testsService.start(1L);
+
+        verify(testsRepository, times(1)).start(any(LocalDateTime.class), anyLong());
```

Comment:
You may not specify `times(1)`, it's used by default.

Reply:
Yes, it's default for a long time. Anyway, specifying times here can make everything clear